src/main.ts

```diff
@@ -138,10 +139,34 @@ function createIdGenerator(): () => Id {
   }
 }
 
-function inferTSConfig(projectPath: string): boolean {
-  // TODO: Move logic from https://github.com/sourcegraph/sourcegraph/pull/30091
-  // into this function.
-  return false;
+/** Returns the list of directory paths under dir which contain a file
+  * with name matchingName.
+*/
+function listDirsRecursive(dir: string, matchingName: string): string[] {
+  let paths: string[] = [];
+  var basenames = fs.readdirSync(dir);
+  for (const basename of basenames) {
+    const path = dir + '/' + basename;
+    const stat = fs.statSync(path);
+    if (stat.isDirectory()) {
+      paths = paths.concat(listDirsRecursive(path, matchingName));
+    } else if (stat.isFile() && basename == matchingName) {
+      paths.push(dir);
+    }
+  }
+  return paths;
+}
+
+function createTSConfigAndEmitLSIF(dir: string): void {
+  // TypeScript packages on NPM are commonly distributed without a
+	// tsconfig.json file. Instead of skipping indexing altogether,
+	// synthesize some permissive tsconfig.json files.
+  fs.writeFileSync(dir + '/' + "tsconfig.json", '{"compilerOptions":{"allowJs":true}}');
+  const newArgv = process.argv.slice();
+  const index = newArgv.indexOf("--inferTSConfig");
+  console.assert(index > -1, "Expected --inferTSConfig to be in argv");
+  newArgv.splice(index, 1);
+  child_process.execSync(newArgv.join(" "));

```

Comment:
What's going on here? It's not clear to me what process were executing

Reply:
It's creating a copy of the original process but with three changes:

1. There will be a minimal `tsconfig.json` (instead of no such file being present).
2. The working directory will be the directory containing the `tsconfig.json`.
3. The argv will be the same as that for the original process, but without the `--inferTSConfig`.

Does that make sense?